.github/workflows/docs.yml

```diff
@@ -8,20 +10,39 @@ on:
     paths:
       - '**/*.md'
       - '.markdownlint-cli2.yaml'
+      - '.github/workflows/docs.yml' # This workflow
   pull_request:
     branches:
       - "main"
       - "release-**"
     paths:
       - '**/*.md'
       - '.markdownlint-cli2.yaml'
+      - '.github/workflows/docs.yml' # This workflow
+
+env:
+  LC_ALL: en_US.UTF-8
+
+defaults:
+  run:
+    shell: bash
+
+permissions:
+  contents: read
 
 jobs:
   markdown-lint:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v4
-      - name: markdownlint-cli2-action
-        uses: DavidAnson/markdownlint-cli2-action@v15
+      - name: "Harden Runner"
+        uses: step-security/harden-runner@17d0e2bd7d51742c71671bd19fa12bdc9d40a3d6 # v2.8.1
+        with:
+          egress-policy: audit # TODO: change to 'egress-policy: block' after couple of runs
```

Comment:
@bjhargrave do you think we could change this since this has run many times?

Reply:
We are just introducing hardening to this repo and the specific egress needs to be seen before we can lock it down. Yes, this workflow is similar to the docs workflow in other repos, but I would like to run it a few times in this repo before we change from `audit`.

Reply:
SGTM